checkov/terraform/plan_runner.py

```diff
@@ -247,14 +248,18 @@ def run_block(self, entities,
                     report.add_record(record=record)
 
     def get_entity_context_and_evaluations(self, entity):
-        raw_context = self.get_entity_context(entity[CustomAttributes.BLOCK_NAME].split("."),
-                                              entity[CustomAttributes.FILE_PATH])
+        entity_id = entity[TF_PLAN_RESOURCE_ADDRESS]
+        raw_context = self.context.get(entity[CustomAttributes.FILE_PATH], {}).get(entity_id)
         if raw_context:
             raw_context['definition_path'] = entity[CustomAttributes.BLOCK_NAME].split('.')
         return raw_context, None
 
-    def get_entity_context(self, definition_path, full_file_path):
-        entity_id = ".".join(definition_path)
+    def get_entity_context(self, definition_path, full_file_path, entity):
+        resource_type = definition_path[0]
+        resource_name = ''
+        if resource_type in entity.keys():
+            resource_name = definition_path[1]
```

Comment:
why do we need this?

Reply:
probably we could simplify it, when the `defintion_path` has a length of at least 2
```python
        resource_type = definition_path[0]
        resource_name = definition_path[1]
```